wait/host_port.go

```diff
@@ -28,28 +28,44 @@ type HostPortStrategy struct {
 	// all WaitStrategies should have a startupTimeout to avoid waiting infinitely
 	timeout      *time.Duration
 	PollInterval time.Duration
+
+	// SkipInternalCheck is a flag to skip the internal check, which is useful when
+	// a shell is not available in the container or when the container doesn't bind
+	// the port internally until additional conditions are met.
+	SkipInternalCheck bool
 }
 
-// NewHostPortStrategy constructs a default host port strategy
+// NewHostPortStrategy constructs a default host port strategy that waits for the given
+// port to be exposed. The default startup timeout is 60 seconds.
 func NewHostPortStrategy(port nat.Port) *HostPortStrategy {
 	return &HostPortStrategy{
 		Port:         port,
 		PollInterval: defaultPollInterval(),
 	}
 }
 
+// ForExposedPortOnly returns a host port strategy that waits for the given port
+// to be exposed only, it does not wait for the port to be bound inside the container.
+func ForExposedPortOnly(port nat.Port) *HostPortStrategy {
```

Comment:
I could see the value in adding a new constructor function, but I'd prefer consistency across the different wait strategies using the fluid builder pattern, something like this:

```go
  wait.ForListeningPort().SkipInternalCheck() // or any other relevant method name
```

I think this would be more consistent. Thoughts?

Reply:
Personally I find that needlessly verbose for a common option, so opted to follow the existing example set by `ForListeningPort` and `ForExposedPort` which admittedly only configure the port.

It could also be a bit confusing because the use case for this is when the port isn't actually listening.

What do you recon?

Reply:
We can probably find a better name:
- ForExternalPort, using the comments you added for the current `Only` function
- ForNotBoundPort
- ForPort
- ...

Reply:
Agreed the `Only` does seem unnecessary, how about just `ForExposedPort` as that matches the existing docker terminology so makes it clear what its looking to achieve?

Reply:
Mmm we already have the `ForExposedPort` that's why I'd add the fluid builder option to the strategy. In v1, we could go with functional options instead of the builder pattern for wait strategies.

Reply:
Yep forgot I added Only for that reason and naming things is hard :) 

Think I'm gonna do a 180, should we do the builder `SkipInternalCheck` and then review the approach for v1?

Reply:
Went for `WithoutInternalCheck` as the `SkipInternalCheck` conflicts with the field name.